test/Driver/single_stack_test.jl

```diff
@@ -0,0 +1,613 @@
+# and `u`in `vars_state_gradient`
+function compute_gradient_argument!(
+    m::BurgersEquation,
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    transform.ρcT = state.ρcT
+    transform.u = state.ρu / state.ρ
+end;
+
+# Specify where in `diffusive::Vars` to store the computed gradient from
+# `compute_gradient_argument!`. Note that:
+#  - `diffusive.μ∇u` is available here because we've specified `μ∇u` in
+#  `vars_state_gradient_flux`
+#  - `∇transform.u` is available here because we've specified `u` in
+#  `vars_state_gradient`
+#  - `diffusive.μ∇u` is built using an anisotropic diffusivity tensor
+function compute_gradient_flux!(
+    m::BurgersEquation,
+    diffusive::Vars,
+    ∇transform::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    diffusive.α∇ρcT = m.α * ∇transform.ρcT
+    diffusive.μ∇u = Diagonal(SVector(m.μh, m.μh, m.μv)) * ∇transform.u
+end;
+
+# Introduce Rayleigh friction towards a target profile as a source.
+# Note that:
+# - Rayleigh damping is only applied in the horizontal by subtracting
+#  the vertical component of momentum from the momentum vector.
+function source!(
+    m::BurgersEquation,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    args...,
+)
+    ẑ = SVector(0, 0, 1)
+    ρ̄ū =
+        state.ρ * SVector(
+            0.5 - 2 * (aux.z - m.zmax / 2)^2,
+            0.5 - 2 * (aux.z - m.zmax / 2)^2,
+            0.0,
+        )
+    ρu_p = state.ρu - ρ̄ū
+    source.ρu -= m.γ * (ρu_p - ẑ' * ρu_p * ẑ)
+end;
+
+# Compute advective flux.
+# Note that:
+# - `state.ρu` is available here because we've specified `ρu` in
+# `vars_state_conservative`
+function flux_first_order!(
+    m::BurgersEquation,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    flux.ρ = state.ρu
+
+    u = state.ρu / state.ρ
+    flux.ρu = state.ρu * u'
+    flux.ρcT = u * state.ρcT
+end;
+
+# Compute diffusive flux (e.g. ``F(μ, u, t) = -μ∇u`` in the original PDE).
+# Note that:
+# - `diffusive.μ∇u` is available here because we've specified `μ∇u` in
+# `vars_state_gradient_flux`
+function flux_second_order!(
+    m::BurgersEquation,
+    flux::Grad,
+    state::Vars,
+    diffusive::Vars,
+    hyperdiffusive::Vars,
+    aux::Vars,
+    t::Real,
+)
+    flux.ρcT -= diffusive.α∇ρcT
+    flux.ρu -= diffusive.μ∇u
+end;
+
+# ### Boundary conditions
+
+# Second-order terms in our equations, ``∇⋅(G)`` where ``G = μ∇u``, are
+# internally reformulated to first-order unknowns.
+# Boundary conditions must be specified for all unknowns, both first-order and
+# second-order unknowns which have been reformulated.
+
+# The boundary conditions for `ρ`, `ρu` and `ρcT` (first order unknown)
+function boundary_state!(
+    nf,
+    m::BurgersEquation,
+    state⁺::Vars,
+    aux⁺::Vars,
+    n⁻,
+    state⁻::Vars,
+    aux⁻::Vars,
+    bctype,
+    t,
+    _...,
+)
+    if bctype == 1 # bottom
```

Comment:
Don't we have a more abstract way of specifying boundary conditions here and below?

Reply:
We'll clean this up eventually. This is what we have to do for now.